### PR TITLE
Add datawire-envoy

### PR DESCRIPTION
**Pull request by wlynch**
This is a fork of envoy with patches applied by Ambassador for their emissary project.
See https://github.com/emissary-ingress/emissary/blob/ac2dc64c6621cd8ec5617f3328544364bdd3fb01/DevDocumentation/ARCHITECTURE.md#envoy

The only modification here from base envoy is the source location. These should try and stay in sync, but I don't know if there's a better way to represent this.

Related: https://github.com/chainguard-dev/image-requests/issues/3494

#### For new package PRs only

- [x] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [x] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

I believe this supports the same upstream version streams as envoy.

### CODE REVIEW

File: datawire-envoy-1.31.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-oci-entrypoint
+    description: Entrypoint for using Envoy in OCI containers
+    dependencies:
+      runtime:
+        - busybox
+      provides:
+        - envoy-oci-entrypoint=${{package.full-version}}
```
Comment (ajayk) [inline datawire-envoy-1.31.yaml:79]: This can't provide envoy-oci-entrypoints needs changes

File: datawire-envoy-1.31.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/var/lib/envoy/init
+          cp envoy/ci/docker-entrypoint.sh ${{targets.subpkgdir}}/var/lib/envoy/init/envoy-entrypoint.sh
+          chmod +x ${{targets.subpkgdir}}/var/lib/envoy/init/envoy-entrypoint.sh
+
+  - name: ${{package.name}}-config
+    description: Default Envoy configuration
+    dependencies:
+      provides:
+        - envoy-config=${{package.full-version}}
```
Comment (ajayk) [inline datawire-envoy-1.31.yaml:90]: This can't provide envoy-config needs changes

File: datawire-envoy-1.31.yaml (hunk 1 above)
Comment (wlynch) [reply datawire-envoy-1.31.yaml]: I think we only need the actual envoy binary and can probably reuse the existing envoy subpackages, so removed.

File: datawire-envoy-1.31.yaml (hunk 2 above)
Comment (wlynch) [reply datawire-envoy-1.31.yaml]: Removed.

**Review by ajayk — APPROVED** [review]

**Pull request merged**